art/attacks/evasion/fast_gradient.py

```diff
@@ -431,20 +431,19 @@ def _apply_norm(norm, grad, object_type=False):
             ).any():
                 logger.info("The loss gradient array contains at least one positive or negative infinity.")
 
+            grad_2d = grad.reshape(1 if object_type else len(grad), -1)
             if norm in [np.inf, "inf"]:
-                grad = np.sign(grad)
+                grad_2d = np.ones_like(grad_2d)
             elif norm == 1:
-                if not object_type:
-                    ind = tuple(range(1, len(x.shape)))
-                else:
-                    ind = None
-                grad = grad / (np.sum(np.abs(grad), axis=ind, keepdims=True) + tol)
-            elif norm == 2:
-                if not object_type:
-                    ind = tuple(range(1, len(x.shape)))
-                else:
-                    ind = None
-                grad = grad / (np.sqrt(np.sum(np.square(grad), axis=ind, keepdims=True)) + tol)
+                i_max = np.argmax(np.abs(grad_2d), axis=1)
+                grad_2d = np.zeros_like(grad_2d)
+                grad_2d[range(len(grad_2d)), i_max] = 1
+            elif norm > 1:
+                q = norm / (norm - 1)
```

Comment:
The variable name `p` and other single character variable names must be replaced with at leaset 3 character names here and all other files to pass the pylint check.

The full pylon report: https://github.com/Trusted-AI/adversarial-robustness-toolbox/actions/runs/8565558666/job/23507533560?pr=2382

Reply:
I think readability takes a *big* toll because of this, but I used what I think were "reasonable" name variables. In the context of $L^p$ spaces, $p$ and $q$ are quite natural variables, that carry a strong meaning. I think it would have been nice to take advantage of that.